scrapyjs/middleware.py

```diff
@@ -96,20 +108,31 @@ def process_request(self, request, spider):
             # are not respected.
             headers=Headers({'Content-Type': 'application/json'}),
         )
-
         self.crawler.stats.inc_value('splash/%s/request_count' % endpoint)
         return req_rep
 
     def process_response(self, request, response, spider):
-        splash_options = request.meta.get("_splash_processed")
+        splash_options = self.get_splash_options(request, spider)
         if splash_options:
             endpoint = splash_options['endpoint']
             self.crawler.stats.inc_value(
                 'splash/%s/response_count/%s' % (endpoint, response.status)
             )
-
+            response = self.html_response(response, request)
         return response
 
+    def html_response(self, response, request):
+        """Give user nice HTML response he probably
+        expects.
+        """
+        data = json.loads(response.body)
+        html = data.get("html")
```

Comment:
maybe `html = data.pop('html', None)`? Is there any reason to keep html in meta in this case?

also it might be good idea to have `'splash': {}` option to enable/disable this behavior, like `'splash': {'transparent': True}`.